### PR TITLE
Ci/update zarr 3.0.9

### PR DESCRIPTION
**Pull request by Faolain**
## Summary by CodeRabbit

* **New Features**
  * Added methods to support toggling read-only mode for HAMT-based Zarr stores, allowing seamless switching between read and write access.
  * Enhanced test coverage for IPFS gateway interactions, including support for DAG-CBOR content and gateway URL variations.

* **Bug Fixes**
  * Enforced read-only restrictions on Zarr stores, preventing write or delete operations when in read-only mode.
  * Improved cleanup and lifecycle management of HTTP clients, ensuring proper resource release in both synchronous and asynchronous contexts.

* **Tests**
  * Expanded and added tests for edge cases in client cleanup, gateway behavior, and read-only store access.
  * Added tests verifying read-only guard behavior for encrypted and plain stores.
  * Updated test utilities to use the latest IPFS Docker image for more reliable integration testing.

* **Chores**
  * Updated IPFS installation in CI workflows and pinned the "zarr" dependency to version 3.0.9 for improved stability.

### CODE REVIEW

**Review by TheGreatAlgo — APPROVED** [review]

File: py_hamt/encryption_hamt_store.py [hunk 1]
```diff
@@ -110,6 +110,19 @@ def __init__(
...
+
+        clone = type(self).__new__(type(self))
+        clone.hamt = self.hamt
+        clone.encryption_key = self.encryption_key
+        clone.header = self.header
+        clone.metadata_read_cache = self.metadata_read_cache
+        clone._forced_read_only = read_only  # safe; attribute is declared
+        zarr.abc.store.Store.__init__(clone, read_only=read_only)
+        return clone
+
```
[the remark this answers is not in the document]
Comment (TheGreatAlgo) [reply py_hamt/encryption_hamt_store.py:125]: damn right it will work

**Review by TheGreatAlgo — APPROVED** [review]
